packages/design-system/src/components/contextMenu/menu.js

```diff
@@ -113,33 +126,64 @@ const Menu = ({
         return;
       }
 
-      const focusableChildren = getFocusableChildren(menuRef.current);
+      const focusableChildren = getFocusableChildren(
+        menuRef.current,
+        isSubMenu
+      );
 
-      const prevIndex = focusableChildren.findIndex(
+      let prevIndex = focusableChildren.findIndex(
         (element) => element.id === focusedId
       );
 
       if (prevIndex === -1 && focusableChildren.length) {
         setFocusedId(focusableChildren[0].id);
-        return;
+        prevIndex = 0;
       }
 
-      const isAscending = [KEYS.UP, KEYS.LEFT].includes(key);
-      let newIndex = prevIndex + (isAscending ? -1 : 1);
+      const prevItemHasSubMenuOpen =
+        focusableChildren[prevIndex]?.hasAttribute('aria-haspopup') &&
+        focusableChildren[prevIndex]?.getAttribute('aria-pressed');
 
-      if (newIndex === -1) {
-        newIndex = focusableChildren.length - 1;
-      }
+      // If we're moving up-down.
+      if ([KEYS.UP, KEYS.DOWN].includes(key)) {
+        const isAscending = KEYS.UP === key;
+        let newIndex = prevIndex + (isAscending ? -1 : 1);
 
-      // Otherwise move to the next element or loop around the list.
-      const newSelectedElement =
-        focusableChildren[newIndex % focusableChildren.length];
+        if (newIndex === -1) {
+          newIndex = focusableChildren.length - 1;
+        }
 
-      newSelectedElement?.focus();
-      setFocusedId(newSelectedElement?.id || -1);
-      return;
+        // Otherwise move to the next element or loop around the list.
+        const newSelectedElement =
+          focusableChildren[newIndex % focusableChildren.length];
+
+        newSelectedElement?.focus();
+        setFocusedId(newSelectedElement?.id || -1);
+        return;
+      }
+      // Maybe move from submenu to parent menu.
+      if (isSubMenu && (KEYS.LEFT === key || (isRTL && KEYS.RIGHT === key))) {
+        // Get the toggled button sharing the same parent.
+        const parentButton = menuRef.current.parentElement.querySelector(
+          'button[aria-pressed="true"][aria-haspopup="menu"]'
+        );
+        parentButton?.focus();
+        return;
+      }
+      // Maybe move into submenu.
+      if (
+        !isSubMenu &&
+        (KEYS.RIGHT === key || (isRTL && KEYS.LEFT === key)) &&
+        prevItemHasSubMenuOpen
+      ) {
+        const subMenu = menuRef.current.querySelector('[role="menu"]');
+        if (subMenu) {
+          const subMenuItems = getFocusableChildren(subMenu, true);
+          subMenuItems[0]?.focus();
+        }
```

Comment:
It's great that you can move between menu and sub menu with the arrow keys. However, for keyboard accessibility I believe it would be best if a user could open the sub menu using the left/right arrow keys as well.

Radix ui has an excellent context menu that I use for reference on keyboard accessibility and component design. https://www.radix-ui.com/docs/primitives/components/context-menu

Reply:
Yep, confirmed with UX that left/right arrow keys should open the menu as well, also the moving out of the menu with left-right arrow key will close the menu.

And should open on hover, clicking on `Select Layer` will not do anything with mouse.

Thanks for the suggestions 👍

Reply:
Just typed this same suggestion in another comment, now deleted it. I agree with all the parts mentioned.

One thing though, when displaying the submenu on hover, make sure to close the menu again only after a short delay on mouseout (e.g. 300ms).

If you're dragging the mouse directly from the parent item to an item further down the submenu list, you might inadvertently pass through other items in the parent list, but you don't actually intend to close the menu because of this.

(I also think the submenu should only open after a short delay)